Add documentation on how to use highlights (#422)
asciidoctor/asciidoctor-reveal.js: docs/modules/converter/pages/syntax/syntax-highlighting.adoc

if you are using the Asciidoctor/Ruby/Bundler toolchain (not Asciidoctor.js/JavaScript/npm).
Check the `examples/` directory for examples and notes about what needs to be done for them to work.
They are considered unsupported by the asciidoctor-reveal.js project.

== Line number highlights

> Reveal.js can add https://revealjs.com/code/#line-numbers-%26-highlights[Line Numbers & Highlights] to source code blocks.

Reviewer: ```suggestion
reveal.js can add https://revealjs.com/code/#line-numbers-%26-highlights[Line Numbers & Highlights] to source code blocks.
```

This example `source` block renders with no line numbers by default:

[source, asciidoc]
-----
[source,java]
----
public class TestClass {
    public TestClass() {

    }

    public void testMethod() {

    }
}
----
-----

> By adding `highlight`, line numbers will be added:

Reviewer: You're right but I think we should recommend to use the `linenums` attribute to enable line numbering.

Author: PR updated for this.

[source, asciidoc]
-----
[source,java,highlight]
----
public class TestClass {
    public TestClass() {

    }

    public void testMethod() {

    }
}
----
-----

> It is also possible to highlight only part of the source. This will highlight the `testMethod()` code:

Reviewer: I believe that this feature is only supported when using Highlight.js (the default source highlighter). Maybe we should add a note, what do you think?

Author: Good idea to add a note for that indeed!

Author: PR updated for this.

[source, asciidoc]
-----
[source,java,highlight=6..8]
----
public class TestClass {
    public TestClass() {

    }

    public void testMethod() {

    }
}
----
-----

NOTE: You can also use  `highlight=6-8` as an alternative to `highlight=6..8`.

> Finally, it is also possible to use https://revealjs.com/code/#step-by-step-highlights[Step-by-step Highlights] by seperating the different highlights with a pipe (`|`) symbol. For example:

Reviewer: ```suggestion
Finally, it is also possible to use https://revealjs.com/code/#step-by-step-highlights[step-by-step Highlights] by separating the different highlights with a pipe (`|`) symbol. For example:
```

if you are using the Asciidoctor/Ruby/Bundler toolchain (not Asciidoctor.js/JavaScript/npm).
Check the `examples/` directory for examples and notes about what needs to be done for them to work.
They are considered unsupported by the asciidoctor-reveal.js project.

== Line number highlights

reveal.js can add https://revealjs.com/code/#line-numbers-%26-highlights[Line Numbers & Highlights] to source code blocks.

This example `source` block renders with no line numbers by default:

[source, asciidoc]
-----
[source,java]
----
public class TestClass {
    public TestClass() {

    }

    public void testMethod() {

    }
}
----
-----

By adding `linenums`, line numbers will be added:

[source, asciidoc]
-----
[source,java,linenums]
----
public class TestClass {
    public TestClass() {

    }

    public void testMethod() {

    }
}
----
-----

> It is also possible to highlight only part of the source by using `highlight`. This will highlight the `testMethod()` code:

Reviewer: (suggestion)

```suggestion
It is also possible to highlight only part of the source by using `highlight`. This will highlight the whole `testMethod()` method:
```

[source, asciidoc]
-----
[source,java,linenums,highlight=6..8]
----
public class TestClass {
    public TestClass() {

    }

    public void testMethod() {

    }
}
----
-----

> WARNING: The highlighting only works using the default source code formatter https://highlightjs.org/[highlight.js].

Reviewer: ```suggestion
WARNING: The highlighting only works using the default source code highlighter https://highlightjs.org/[Highlight.js].
```

> NOTE: You can also use  `highlight=6-8` as an alternative to `highlight=6..8`.

Reviewer: ```suggestion
NOTE: You can also use `highlight=6-8` as an alternative to `highlight=6..8`.
```
